filament/src/View.cpp

```diff
@@ -505,18 +505,23 @@ void FView::prepare(FEngine& engine, backend::DriverApi& driver, ArenaScope& are
 
         /*
          * Shadowing: compute the shadow camera and cull shadow casters
-         * (this will set the VISIBLE_DIR_SHADOW_CASTER bit)
+         * (this will set the VISIBLE_DIR_SHADOW_CASTER bit and VISIBLE_SPOT_SHADOW_CASTER bits)
          */
 
         prepareShadowing(engine, driver, renderableData, scene->getLightData());
 
         /*
-         * partition the array of renderable w.r.t their visibility:
+         * Parition the SoA so that renderables are partitioned w.r.t their visibility into the
+         * following groups:
          *
-         * Sort the SoA so that invisible objects are first, then renderables,
-         * then both renderable and casters, then casters only -- this operation is somewhat heavy
-         * as it sorts the whole SoA. We use std::partition instead of sort(), which gives us
-         * O(3.N) instead of O(N.log(N)) application of swap().
+         * - renderables
+         * - renderables and directional shadow casters
+         * - directional shadow casters only
```

Comment:
I think this is misleading because this bucket can contain "spot shadow casters" now. In fact all buckets can.

Reply:
Great point